src/controller/python/chip/commissioning/commissioning_flow_blocks.py

```diff
@@ -0,0 +1,182 @@
+
+import logging
+
+from . import pase
+
+
+class CommissioningFlowBlocks:
+    def __init__(self, devCtrl: ChipDeviceCtrl.ChipDeviceControllerBase, credential_provider: commissioning.CredentialProvider, logger: logging.Logger):
+        self._devCtrl = devCtrl
+        self._logger = logger
+        self._credential_provider = credential_provider
+
+    async def arm_failsafe(self, parameter: commissioning.PaseParameters, device: pase.Session):
+        response = await self._devCtrl.SendCommand(device.node_id, commissioning.ROOT_ENDPOINT_ID, Clusters.GeneralCommissioning.Commands.ArmFailSafe(
+            expiryLengthSeconds=parameter.failsafe_expiry_length_seconds
+        ))
+        if response.errorCode != 0:
+            raise commissioning.CommissionFailure(repr(response))
+
+    async def operational_credentials_commissioning(self, parameter: commissioning.Parameters, device: pase.Session):
+        self._logger.info("Getting Remote Device Info")
+        device_info = (await self._devCtrl.ReadAttribute(device.node_id, [
+            (commissioning.ROOT_ENDPOINT_ID, Clusters.BasicInformation.Attributes.VendorID),
+            (commissioning.ROOT_ENDPOINT_ID, Clusters.BasicInformation.Attributes.ProductID)], returnClusterObject=True))[commissioning.ROOT_ENDPOINT_ID][Clusters.BasicInformation]
+
+        self._logger.info("Getting AttestationNonce")
+        attestation_nonce = await self._credential_provider.get_attestation_nonce()
+
+        self._logger.info("Sending AttestationRequest")
+        attestation_elements = await self._devCtrl.SendCommand(device.node_id, commissioning.ROOT_ENDPOINT_ID, Clusters.OperationalCredentials.Commands.AttestationRequest(
+            attestationNonce=attestation_nonce
+        ))
+
+        self._logger.info("Getting CertificateChain - DAC")
+        dac = await self._devCtrl.SendCommand(device.node_id, commissioning.ROOT_ENDPOINT_ID, Clusters.OperationalCredentials.Commands.CertificateChainRequest(
+            certificateType=1
+        ))
+
+        self._logger.info("Getting CertificateChain - PAI")
+        pai = await self._devCtrl.SendCommand(device.node_id, commissioning.ROOT_ENDPOINT_ID, Clusters.OperationalCredentials.Commands.CertificateChainRequest(
+            certificateType=2
+        ))
+
+        self._logger.info("Getting CSR Nonce")
+        csr_nonce = await self._credential_provider.get_csr_nonce()
+
+        self._logger.info("Getting OpCSRRequest")
+        csr = await self._devCtrl.SendCommand(device.node_id, commissioning.ROOT_ENDPOINT_ID, Clusters.OperationalCredentials.Commands.CSRRequest(
+            CSRNonce=csr_nonce
+        ))
+
+        self._logger.info("Getting device certificate")
+        commissionee_credentials = await self._credential_provider.get_commissionee_credentials(
+            commissioning.GetCommissioneeCredentialsRequest(
+                dac=dac, pai=pai,
+                attestation_challenge=attestation_nonce,
```

Comment:
attestation_challenge comes from secure session, NOT nonces. 

We need both attestation_nonce and attestation_challenge. 

Attestation challenge is gotten from:

```
    // Retrieve attestation challenge
    ByteSpan attestationChallenge =
        proxy->GetSecureSession().Value()->AsSecureSession()->GetCryptoContext().GetAttestationChallenge();
```

We need a way to access that from Python